### PR TITLE
[APL-1007] Refactor the current application to support H2 1.4.200

### PR DESCRIPTION
**Pull request by silaev-firstbridge**
This PR breaks backward compatibility with the current MV_STORE=false approach.
Therefore, do not merge this PR until regression testing is done and a plan to migrate is set

In case running tests, please, use `mvn -Dh2.version=1.4.199-2060 clean install`

### CODE REVIEW

File: apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/db/ShardInitTableSchemaVersion.java [hunk 1]
```diff
@@ -15,7 +15,7 @@ protected int update(int nextUpdate) {
                         "DB_ID BIGINT not null, " +
                         "ID BIGINT not null, " +
                         "VERSION INTEGER not null, " +
-                        "TIMESTAMP INTEGER not null, " +
+                        "\"TIMESTAMP\" INTEGER not null, " +
```
Comment (AndrewBoyarsky) [inline apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/db/ShardInitTableSchemaVersion.java]: Same field is also present in AplDbVersion

File: apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/db/ShardRecoveryDaoJdbcImpl.java [hunk 2]
```diff
@@ -144,7 +145,8 @@ public long saveShardRecovery(Connection con, ShardRecovery recovery) {
         try (PreparedStatement pstmt = con.prepareStatement(
                 "INSERT INTO shard_recovery(" +
                         "state, object_name, column_name, last_column_value, processed_object, updated, height) " +
-                        "VALUES (?, ?, ?, ?, ?, CURRENT_TIMESTAMP(), ?)"
+                        "VALUES (?, ?, ?, ?, ?, CURRENT_TIMESTAMP(), ?)",
```
Comment (AndrewBoyarsky) [inline apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/db/ShardRecoveryDaoJdbcImpl.java]: Does New h2 version not return generated keys without additional parameter?

File: apl-core/src/main/java/com/apollocurrency/aplwallet/apl/exchange/dao/DexOrderDao.java [hunk 3]
```diff
@@ -34,7 +34,7 @@
             "AND (:offerCur is NULL OR offer.offer_currency = :offerCur) " +
             "AND (:pairCur is NULL OR offer.pair_currency = :pairCur) " +
             "ORDER BY offer.pair_rate DESC " +
-            "OFFSET :offset LIMIT :limit"
+            "OFFSET :offset FETCH FIRST :limit ROWS ONLY"
```
Comment (AndrewBoyarsky) [inline apl-core/src/main/java/com/apollocurrency/aplwallet/apl/exchange/dao/DexOrderDao.java]: Standard limit is not working? In this doc limit can be used `http://www.h2database.com/html/commands.html#select`

File: apl-core/src/test/resources/account_control_phasing.csv [hunk 4]
```diff
@@ -1,5 +1,5 @@
 ACCOUNT_ID(-5|19|0),WHITELIST(2003|0|0),VOTING_MODEL(-6|3|0),QUORUM(-5|19|0),MIN_BALANCE(-5|19|0),HOLDING_ID(-5|19|0),MIN_BALANCE_MODEL(-6|3|0),MAX_FEES(-5|19|0),MIN_DURATION(5|5|0),MAX_DURATION(5|5|0),HEIGHT(4|10|0)
 7995581942006468815, null, 0, 1, null, null, 0, 300000000, 12, 113, 500
-2728325718715804811,(-8446656647637444484,),0,1,null,null,0,300000000,12,113,1000
--8446384352342482748,(-8446656647637444484,1344527020205736624,),0,1,null,null,0,300000000,12,113,2000
--4013722529644937202,(-8446656647637444484,1344527020205736624,-6724281675870110558,),0,1,null,null,0,300000000,12,113,3000
+2728325718715804811,(-8446656647637444484),0,1,null,null,0,300000000,12,113,1000
```
Comment (AndrewBoyarsky) [inline apl-core/src/test/resources/account_control_phasing.csv]: Comma was left here due to h2 bug in Parser, it forces h2 to interpret single value as array. Is it alredy fixed?

File: apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/db/ShardRecoveryDaoJdbcImpl.java (hunk 2 above)
Comment (silaev-firstbridge) [reply apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/db/ShardRecoveryDaoJdbcImpl.java]: nope

File: apl-core/src/test/resources/account_control_phasing.csv (hunk 4 above)
Comment (silaev-firstbridge) [reply apl-core/src/test/resources/account_control_phasing.csv]: now this (-8446656647637444484,1344527020205736624,) throws an exception, so the last comma must not be there

File: apl-core/src/main/java/com/apollocurrency/aplwallet/apl/exchange/dao/DexOrderDao.java (hunk 3 above)
Comment (silaev-firstbridge) [reply apl-core/src/main/java/com/apollocurrency/aplwallet/apl/exchange/dao/DexOrderDao.java]: Even though LIMIT is supposed to work, this "OFFSET :offset LIMIT :limit" leads to 
`Caused by: org.jdbi.v3.core.statement.UnableToCreateStatementException: org.h2.jdbc.JdbcSQLSyntaxErrorException: Syntax error in SQL statement "SELECT * FROM DEX_OFFER AS OFFER WHERE LATEST = TRUE AND (OFFER.DB_ID > ?) AND (? IS NULL OR OFFER.ACCOUNT_ID = ?) AND (? IS NULL OR OFFER.FINISH_TIME > ?) AND (? IS NULL OR OFFER.TYPE = ?) AND (? IS NULL OR OFFER.STATUS = ?) AND (? IS NULL OR OFFER.OFFER_CURRENCY = ?) AND (? IS NULL OR OFFER.PAIR_CURRENCY = ?) ORDER BY OFFER.PAIR_RATE DESC OFFSET ? LIMIT[*] ?"`

File: apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/db/ShardInitTableSchemaVersion.java (hunk 1 above)
Comment (silaev-firstbridge) [reply apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/db/ShardInitTableSchemaVersion.java]: no need to do it there

Comment (silaev-firstbridge) [reply apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/db/ShardInitTableSchemaVersion.java]: done

File: apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/account/LedgerEntry.java [hunk 5]
```diff
@@ -255,7 +255,10 @@ public boolean equals(Object obj) {
      * @throws  SQLException            Database error occurred
      */
     void save(Connection con) throws SQLException {
-        try (final PreparedStatement stmt = con.prepareStatement("INSERT INTO account_ledger " + "(account_id, event_type, event_id, holding_type, holding_id, change, balance, " + "block_id, height, timestamp) " + "VALUES(?, ?, ?, ?, ?, ?, ?, ?, ?, ?)", Statement.RETURN_GENERATED_KEYS)) {
+        try (final PreparedStatement stmt = con.prepareStatement(
```
Comment (yuriy-larin-firstbridge) [inline apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/account/LedgerEntry.java]: Is annotation
@DatabaseSpecificDml(DmlMarker
is missing here ?

File: apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/app/TrimDao.java [hunk 6]
```diff
@@ -33,7 +34,10 @@ public TrimEntry save(TrimEntry trimEntry) {
                     return trimEntry;
                 }
             } else {
-                try (PreparedStatement pstm = con.prepareStatement("INSERT INTO trim (height, done) VALUES (?, ?) ")) {
+                try (PreparedStatement pstm = con.prepareStatement(
```
Comment (yuriy-larin-firstbridge) [inline apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/app/TrimDao.java]: Is missing 
@DatabaseSpecificDml(DmlMarker..... )
?

File: apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/account/LedgerEntry.java (hunk 5 above)
Comment (silaev-firstbridge) [reply apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/account/LedgerEntry.java]: this annotation marks only SQL that is specific to DB. INSERT is common for all RDBMS

File: apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/app/TrimDao.java (hunk 6 above)
Comment (silaev-firstbridge) [reply apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/app/TrimDao.java]: same as above

**Pull request closed without merging**